test/integration/query-interface/schemas.test.ts

```diff
@@ -36,4 +39,48 @@ describe('QueryInterface#createSchema', async () => {
 
     expect(queryGeneratorSpy.args[0]).to.include(options);
   });
+
+  it('drops a schema', async () => {
+    await queryInterface.createSchema(testSchema);
+    const preDeletionSchemas = await queryInterface.showAllSchemas();
+    expect(preDeletionSchemas).to.include(testSchema, 'createSchema did not create testSchema');
+
+    await queryInterface.dropSchema(testSchema);
+    const postDeletionSchemas = await queryInterface.showAllSchemas();
+    expect(postDeletionSchemas).to.not.include(testSchema, 'dropSchema did not drop testSchema');
+  });
+
+  it('shows all schemas', async () => {
+    await queryInterface.createSchema(testSchema);
+    const allSchemas = await queryInterface.showAllSchemas();
+
+    const expected = dialectsWithWeirdSchemas.includes(dialect.name)
+      ? [testSchema]
+      : [sequelize.config.database, testSchema];
+    expect(allSchemas.sort()).to.deep.eq(expected.sort());
+  });
+
+  describe('drops all schemas', () => {
+    it('drops all schemas except test schema', async () => {
+      await queryInterface.dropAllSchemas({
+        skip: [sequelize.config.database],
+      });
+      const postDeleteSchemas = await queryInterface.showAllSchemas();
+
+      const expected = dialectsWithWeirdSchemas.includes(dialect.name) ? [] : [sequelize.config.database];
```

Comment:
We're not actually testing the skip option for postgres, mssql and db2 with this. We could give them their own test in which we drop everything but `testSchema`